main/src/library/MutDeque.flix

```diff
@@ -357,4 +357,26 @@ namespace MutDeque {
         foldLeft((_, x) -> { append!(sb, f(x)); append!(sb, sep) }, (), d);
         StringBuilder.toString(sb) |> String.dropRight(lastSep)
 
+    ///
+    /// Apply the effectful function `f` to all the elements in the MutDeque `d`.
+    ///
+    pub def foreach(f: a -> Unit & ef, d: MutDeque[a]): Unit & Impure =
+        let MutDeque(a, front, back) = d;
+        let len = Array.length(deref a);
+
+        def loop(i) = {
+            if (i == deref back) {
+                ()
+            } else {
+                f(Array.get(i, deref a));
+                loop((i + 1) &&& (len - 1))
```

Comment:
Thanks to @jaschdoc for the bit `&&&` hint

Reply:
Maybe it's ugly but it's fast :joy: